### DISCUSSION TITLE
src/run/salmon.mk     info.json file had to be renamed to refInfo.json

### DISCUSSION THREAD

**Discussion by njbowen** [biostars/biostar-handbook-issues#324] [General]
in the current src/run/salmon.mk     file
I had to rename info.json on line 40 to refInfo.json to get the salmon.mk to find the index files.

**Comment by ialbert** [comment]
thanks for the report, might be that they renamed the index file, I will investigate

**Comment by ialbert** [comment]
Which version of salmon is this, I tried with version 1.10.3 and the index file was still info.json

**Comment by njbowen** [comment]
$ salmon -v
salmon 0.14.1
(bioinfo) 

$which salmon
/Users/myusername/micromamba/envs/bioinfo/bin/salmon

**Reply by ialbert** [reply]
Ah yes, this is well known annoyance with salmon library dependency incompatibility, it has to roll itself back to an extraordinarily old version

that salmon is very-very outdated, the current version is 1.10

install salmon in a brand new environment where it "lives" alone then run from the main `bioinfo` environment as:

    micromamba run -n salmonenv  salmon

I should detect the version of salmon in the makefile and inform people of this issue

**Reply by njbowen** [reply]
Will do, many thanks.

**Reply by njbowen** [reply]
point of clarification, if I'm running src/workflows/benchmark.mk airway just substituting the micromamba run -n salmonenv salmon in the salmon.mk file where salmon is called will not work correct, i have to follow the "switching" envs in a script instructions you give?

**Reply by ialbert** [reply]
You would need the makefile in the salmonenv for the salmon specific target, I

    micromamba run -n salmonenv make align

if that is not possible, for example the makefile will trigger other steps then it needs to be edited to run the salmon specific task as above. You can think of 

   micromamba run -n salmonenv 
   
as a temporary switch of environments   
  
that being said, if I recall correctly the main culprit for the salmon downgrade might be R version 4, once an environment has R 4.0 salmon (and other tools) won't play well and have to be downgraded to a very old versions

So a solution might be to separate the statistics out of bioinfo

The latest versions of the book I recommend doing that. Basically one ends up with two environments a `bioinfo` and a `stats` env. Here is described:

https://www.biostarhandbook.com/books/workflows/help/statsenv/

I am planning to  modify the install script to set up both environments right at the start and to teach people to switch environments.

**Reply by njbowen** [reply]
Yes, we’ve used the stats env as well.  But it installs R 4.3 I believe.  I suppose that I could set up an env and try to install a previous version of R there, if that’s the salmon issue.

**Reply by ialbert** [reply]
Since Bioconductor is tied to R>=4, we cannot install a previous version of R unfortunately.

**Reply by njbowen** [reply]
Ok, thanks, salmon is not an R package, so I can make something work in a separate env, correct?

**Reply by ialbert** [reply]
yes, I found that salmon works best if it lives in its own environment. It makes use of  C++ libraries that are also used in numerous other software - and with that creates a lot of conflicts.

best to run salmon from its own env which is not much trouble, though a bit annoying as the command line grows:

    micromamba run -n salmonenv salmon

**Reply by njbowen** [reply]
Yes, full circle, you did recommended that initially above.  I should remember to scroll up! Thank you.
